server/src/main/java/io/spine/server/bus/Bus.java

```diff
@@ -61,6 +63,15 @@
     private DispatcherRegistry<C, D> registry;
 
     /**
+     * The chain of filters for this bus.
+     *
+     * <p>This field is effectively final, but is initialized lazily.
+     *
+     * @see #filterChain() for the non-null filter chain value
+     */
+    private FilterChain<E, T> filterChain;
```

Comment:
Theoretically, `E` may be or be not bounded by `T`. To ensure we have the same message type, not `MessageEnvelope<?>`, we introduce the second type parameter here.